### ISSUE TITLE
desktop-base: a more simplified model for more pleasant DE installation

### ISSUE THREAD

**Issue by MingcongBai** [AOSC-Dev/aosc-os-abbs#292]
Currently for "scientific" or "correctness", when installing a desktop environment on AOSC OS, some of the most important packages for a proper desktop experience are not installed:

- `xorg-server` for the X11 display server to actually exist.
- `xf86-*` for the X11 display server to actually start (drivers yo).
- Fonts, or else nothing displays (and possibly crashes some DEs).

Debian has a package named `desktop-base` which provides data files essential for DE to be installed properly, but in this case, we should be providing something similar, but serving a different purpose, take this dependency "grid" for example:

|DE: GNOME, KDE, XFCE, etc.                                                |
------------------------------------------------------------------------------
|desktop-base: depends on X11 Server and default fonts |

Simply solves the first and the third issue listed above. The second though is what comes to the question, should `xorg-server` depend on a set of dependencies, like all video and input drivers?

- Video drivers should all be installed, because you will never know...
- Input driver, `xf86-video-libinput` should have a pretty safe coverage on most input devices, perhaps adding `xf86-input-joystick` and `xf86-input-wacom` will be even safer? Oh yes, and `xf86-input-vmmouse`.

My proposed resolution comes that:

- All desktop environment should have a meta-package, like `gnome`, `kde`, and `mate`.
- DE meta-packages depends on all listed above.

So an example will be like.

```
Package: gnome
Section: gnome
Depends: desktop-base, gnome-shell, gdm, nautilus, ...
Description: A easy-to-use, internationalized, and accessible desktop environment
````

```
Package: desktop-base
Section: x11
Depends: xf86-video-intel, xf86-video-amdgpu, ..., xf86-input-libinput, xf86-input-vmmouse, noto-fonts, ...
Description: Base desktop meta-package
```

What do you guys think?

**Comment by Icenowy** [comment]
Have you considered BSPs for ARM?

**Comment by MingcongBai** [comment]
> Have you considered BSPs for ARM?

Still, it's do-able, as we still have the same package names don't we?

Also, a fix in the `desktop-base` sample mock-up.

```
Package: desktop-base
Section: x11
Depends: xorg-server, noto-fonts
Description: Base desktop meta-package
```
And a split onto `xorg-server` below:

```
Package: xorg-server
Section: x11
Depends: ..., xf86-video-intel, xf86-video-amdgpu, ..., xf86-input-libinput, ...
Description: X.Org display server
```

And of course, architectural-based dependencies will be in place.

**Comment by Icenowy** [comment]
xorg-server depends on xf86-*... serious?

**Comment by MingcongBai** [comment]
> xorg-server depends on xf86-*... serious?

Well, or nothing gets installed, that would be an issue...

**Comment by Icenowy** [comment]
It may be better to use another metapackage, or here will be a circuit disaster.
2016年7月21日 11:07于 Mingcong Bai <notifications@github.com>写道：
xorg-server depends on xf86-*... serious?


Well, or nothing gets installed, that would be an issue...

**Comment by MingcongBai** [comment]
> It may be better to use another metapackage, or here will be a circuit disaster.

... Which `desktop-base` depends on?

**Comment by colin4124** [comment]
Some special purpose drivers like`xf86-input-joystick` `xf86-input-wacom`, and `xf86-input-vmmouse`  should not be in `desktop-base`. `desktop-base` should be base as its name sounds.

Considering the user can use aosc out of box in all the case, special purpose drivers should be in `AOSC destop` tarball. 

`AOSC destop` tarball should include all packages related to desktop base as possible. All kinds of user can use aosc out of box in all the case.

`desktop-base` package should be as general as  possible. It should include just **base** packages, for general purpose in common case.

**Comment by MingcongBai** [comment]
> desktop-base package should be as general as possible. It should include just base packages, for general purpose in common case.

Agreed, however, we have no control over the "base" video drivers as people use all kinds of graphics cards, which all requires different drivers (at lease before `xf86-video-modesettings` is perfectly usable).

As to input drivers, `libinput` should be good for our purpose, so let's do that.

**Comment by colin4124** [comment]
Some special purpose drivers like`xf86-input-joystick` `xf86-input-wacom`, and `xf86-input-vmmouse`  should not be in `desktop-base`. `desktop-base` should be base as its name sounds.

Considering the user can use aosc out of box in all the case, special purpose drivers should be in `AOSC destop` tarball. 

`AOSC destop` tarball should include all packages related to desktop base as possible. All kinds of user can use aosc out of box in all the case.

`desktop-base` package should be as general as  possible. It should include just **base** packages, for general purpose in common case.

**Comment by MingcongBai** [comment]
(Removed a dup-comment from @colin4124 )

**Comment by lmy441900** [comment]
It's really unnecessary for someone to have useless graphical card (or whatever) drivers on his/her computer, although we do it in tarballs.

So, let users to install graphical card drivers by themselves. Someone intends to install a DE by himself should know grapical card driver things.

**Comment by MingcongBai** [comment]
> It's really unnecessary for someone to have useless graphical card (or whatever) drivers on his/her computer, although we do it in tarballs.

Yes but a they are all pretty small, except when they are proprietary like `nvidia`. Take `xf86-video-intel` for example.

```
Package: xf86-video-intel
Status: install ok installed
Section: x11
Installed-Size: 2260
Maintainer: Jeff Bai <jeffbai@aosc.xyz>
Architecture: amd64
Version: 1:20160704
Depends: mesa (>= 11.2.2), pixman (>= 0.34.0), xcb-util (>= 0.4.0-0), xorg-server (>= 1.18.3)
Description: XF86 driver for Intel GPUs
```

A mere 2MB for one of the largest XF86 driver in the repository.

> So, let users to install graphical card drivers by themselves.

@colin4124 Who plays around i3 WM and other hacky stuff can find himself not remembering to do so. This is not a matter of knowing a thing or not, it's just really annoying when you find yourself not remembering something while configuring AOSC OS for daily work.

**Comment by lmy441900** [comment]
> Yes but a they are all pretty small

Well if so...

> 结果发现xf86-video-fbturbo没装
> _-- @chenx97_

Maybe you're right @MingcongBai :new_moon_with_face:

**Comment by LionNatsu** [comment]
Why not create some ... `input-drivers` `video-drivers` `printer-drivers`?

Group is good, I hope every pm have one.

**Comment by MingcongBai** [comment]
> Why not create some ... input-drivers video-drivers printer-drivers?

I am willing to do that, but that will be another issue... Generalizing meta packages can be beneficial, but overdoing it may just be problematic.
